### PR TITLE
Deploy the operator in dedicated (openebs) namespace

### LINKED ISSUE #1244: openebs provisioner should not set the namespace to default
The current logic of fetching is shown below:
```go
namespace := os.Getenv("OPENEBS_NAMESPACE")
if namespace == "" {
	namespace = "default"
}
```
However, it will be ideal to use the namespace under which the provisioner is running than setting it to default. 
Refer to [K8s Downward API](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/)

In production, users will not be using default namespace.  Users will have a dedicated namespace & service account for openebs components i.e. maya api server, openebs provisioner, maya api server service, and future ones.

### PR DESCRIPTION
**Pull request by ksatchit**
Signed-off-by: ksatchit <karthik.s@cloudbyte.com>

**What this PR does / why we need it**:

Following changes have been made in the openebs-operator.yaml as part of this commit

- Add spec to create openebs namespace in the operator
- Update the resources' namespace property to openebs
- Add OPENEBS_NAMESPACE ENV variable to the provisioner using K8s downward api
- Remove namespace reference in the clusterRole (clusterRoles by definition is applicable across namespaces) 
- Refactor comments on the manifest 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1244 

**Special notes for your reviewer**:

- Since the deployment yaml has the namespace creation spec included, upgrades are recommended via `strategic patch merge` instead of re-apply of the operator manifest (which might throw an error saying namespace exists) 

- A sample application (mongo-statefulset) created successfully after deploying the updated operator YAML. 

```

karthik@MayaMaster:~$ kubectl get pods -n openebs
NAME                                           READY     STATUS    RESTARTS   AGE
maya-apiserver-6577687899-4dnwp                1/1       Running   0          1h
openebs-provisioner-c86d4f596-dnwhk            1/1       Running   0          1h
openebs-snapshot-controller-586557744f-tb4kf   2/2       Running   0          1h

karthik@MayaMaster:~$ kubectl get pods -n mongo
NAME                                                             READY     STATUS    RESTARTS   AGE
mongo-0                                                          2/2       Running   0          1h
mongo-1                                                          2/2       Running   0          1h
mongo-2                                                          2/2       Running   0          1h
pvc-0d0e0877-6e35-11e8-8136-000c298ff5fc-ctrl-577677d4f6-l4rqn   2/2       Running   0          1h
pvc-0d0e0877-6e35-11e8-8136-000c298ff5fc-rep-5f9c7b8b7d-pmbgs    1/1       Running   0          1h
pvc-28eb42f8-6e35-11e8-8136-000c298ff5fc-ctrl-57d5cf7c94-5gslt   2/2       Running   0          1h
pvc-28eb42f8-6e35-11e8-8136-000c298ff5fc-rep-74896585cb-s6jlq    1/1       Running   0          1h
pvc-be2f94ff-6e34-11e8-8136-000c298ff5fc-ctrl-559f9d64bf-52dvh   2/2       Running   0          1h
pvc-be2f94ff-6e34-11e8-8136-000c298ff5fc-rep-797766669f-m9d42    1/1       Running   0          1h
```

### CODE REVIEW

File: k8s/openebs-operator.yaml [hunk 1]
```diff
@@ -155,6 +159,10 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: spec.nodeName
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
```
Comment (prateekpandey14) [inline k8s/openebs-operator.yaml]: We have to add Namespace downward API to snapshot-controller and snapshot-provisioner deployment containers too.

File: k8s/openebs-operator.yaml [hunk 2]
```diff
@@ -1,23 +1,27 @@
+# Create OpenEBS namespace
 # Define the Service Account
 # Define the RBAC rules for the Service Account
 # Launch the maya-apiserver ( deployment )
 # Launch the maya-storagemanager ( deameon set )
```
Comment (kmova) [inline k8s/openebs-operator.yaml]: @ksatchit - Probably fix the above line as well, though not part of this commit.

Comment (kmova) [reply k8s/openebs-operator.yaml]: Since the following lines will anyway be self descriptive. We can probably put a simple note saying openebs control plane components and associated CRs and RBAC rules are applied

Comment (ksatchit) [reply k8s/openebs-operator.yaml]: @kmova , have made the suggested changes.
